lib/telemetry.go

```diff
@@ -349,6 +352,11 @@ func configureSinks(cfg TelemetryConfig, memSink metrics.MetricSink) (metrics.Fa
 	addSink(dogstatdSink)
 	addSink(circonusSink)
 	addSink(prometheusSink)
+	for _, sink := range cfg.ExtraSinks {
```

Comment:
could refactor this into a `addExtraSinks` function if that's easier to read.

Reply:
Yeah, because we are adding this general use "ExtraSinks", someone else could use this later on to append a sink too. 

If they don't validate for `nil`, it would lead to nil pointer errors later on when metrics get sent. 

For this PR, we do validate that the sink is not `nil` https://github.com/hashicorp/consul/pull/17162/files#diff-92974f7ecae0d7201c3bd5ad4a8e8b0e49a66e40ee72d1661644fbc3cee199c1R108. So that would never occur.

I initially had this new config field as  a single `HCPSink` of type `OTELSink`, and @nickethier suggested a more generic approach.

If we want to limit our changes, we could also make this field a single sink of the interface type `MetricSink` for our purposes.

Let me know what you think!

Reply:
Yeah there _shouldn't_ be a situation where a `nil` sink exists but I don't mind the extra protection here.